### PR TITLE
`TxContext.hash()`

### PR DESCRIPTION
**Pull request by dbanks12**
Blocked by https://github.com/AztecProtocol/aztec3-circuits/pull/1 and https://github.com/AztecProtocol/aztec3-circuits/pull/2

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  AfterStruct: false
+  AfterUnion: false
+  AfterExternBlock: false
+  BeforeCatch: false
+  BeforeElse: false
+  SplitEmptyFunction: false
+  SplitEmptyRecord: false
+  SplitEmptyNamespace: false
+AllowShortFunctionsOnASingleLine : Inline
+SortIncludes: false
```
Comment (iAmMichaelConnor) [inline .clang-format:28]: The crypto team might have made edits to this during their Mexico offsite. Do we want to align with their views?

**Comment by iAmMichaelConnor** [comment]
Looks like a lot more stuff than the PR description? Is this due to stacking PRs?

**Comment by dbanks12** [comment]
> Looks like a lot more stuff than the PR description? Is this due to stacking PRs?

Yeah this doesn't look right....

File: .clang-format (hunk 1 above)
Comment (dbanks12) [reply .clang-format:28]: Yes. This diff shouldn't be in this PR. Something weird is going on with PRs in this repo....

**Comment by dbanks12** [comment]
@iAmMichaelConnor this PR has been fixed. I think it was based off of master before suyash squashed master commits.

**Review by suyash67 — APPROVED** [review]
Reviewed with David, looks good to me.

**Pull request merged**